lib/firmata.js

```diff
@@ -1690,6 +1718,318 @@ Board.prototype.pingRead = function(opts, callback) {
+  }
+
+  data.push(pinsToInvert);
+  data.push(END_SYSEX);
+  this.transport.write(new Buffer(data));
+
+};
+
+/**
+ * Asks the arduino to set the stepper position to 0
+ * Note: This is not a move. We are setting the current position equal to zero
+ * @param {number} deviceNum Device number for the stepper (range 0-9)
+ */
+
+Board.prototype.accelStepperZero = function(deviceNum) {
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x01, // STEPPER_ZERO from firmware
+    deviceNum,
+    END_SYSEX
+  ];
+
+  this.transport.write(new Buffer(data));
+};
+
+/**
+ * Asks the arduino to move a stepper a number of steps
+ * (and optionally with and acceleration and deceleration)
+ * speed is in units of steps/sec
+ * @param {number} deviceNum Device number for the stepper (range 0-5)
+ * @param {number} steps Number of steps to make
+ */
+Board.prototype.accelStepperStep = function(deviceNum, steps, callback) {
+  
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x02, // STEPPER_STEP from firmware
+    deviceNum
+  ];
+
+  Array.prototype.push.apply(data, Board.encode32BitSignedInteger(steps));
+  data.push(END_SYSEX);
+
+  this.transport.write(new Buffer(data));
+  
+  if (callback) {
+    this.once("stepper-done-" + deviceNum, callback);
+  }
+
+};
+
+/**
+ * Asks the arduino to move a stepper to a specific location
+ * @param {number} deviceNum Device number for the stepper (range 0-5)
+ * @param {number} position Desired position
+ */
+Board.prototype.accelStepperTo = function(deviceNum, position, callback) {
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x03, // STEPPER_TO from firmware
+    deviceNum
+  ];
+
+  Array.prototype.push.apply(data, Board.encode32BitSignedInteger(position));
+  data.push(END_SYSEX);
+
+  this.transport.write(new Buffer(data));
+
+  if (callback) {
+    this.once("stepper-done-" + deviceNum, callback);
+  }
+
+};
+
+/**
+ * Asks the arduino to enable/disable a stepper
+ * @param {number} deviceNum Device number for the stepper (range 0-9)
+ * @param {boolean} [enabled]
+ */
+
+Board.prototype.accelStepperEnable = function(deviceNum, enabled) {
+
+  if (typeof enabled === "undefined") {
+    enabled = true;
+  }
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x04, // ENABLE from firmware
+    deviceNum,
+    enabled & 0x01,
+    END_SYSEX
+  ];
+
+  this.transport.write(new Buffer(data));
+};
+
+/**
+ * Asks the arduino to stop a stepper
+ * @param {number} deviceNum Device number for the stepper (range 0-9)
+ */
+
+Board.prototype.accelStepperStop = function(deviceNum) {
+
+  var data = [
+    START_SYSEX,
+    ACCELSTEPPER,
+    0x05, // STEPPER_STOP from firmware
+    deviceNum,
+    END_SYSEX
+  ];
+
+  this.transport.write(new Buffer(data));
+
+};
+
+/**
+ * Asks the arduino to report the position of a stepper
+ * @param {number} deviceNum Device number for the stepper (range 0-9)
+ */
+
+Board.prototype.accelStepperReportPosition = function(deviceNum) {
```

Comment:
Please add an optional callback to this method so the user doesn't have to worry about listening for "stepper-position-N".